### PR TITLE
EVG-14115: Test results support group id

### PR DESCRIPTION
**Pull request by julianedwards**
JIRA: https://jira.mongodb.org/browse/EVG-14115

Test results and attach.results command now support group ids, I will update the wiki once this is merged. I also revendored juniper and timber.

### CODE REVIEW

**Review by Kimchelly — APPROVED** [review]

**Comment by julianedwards** [comment]
I am going to wait until Monday to merge this cause I have to deploy cedar first.

**Pull request merged**
